pkg/kubectl/resource_printer.go

```diff
@@ -1505,6 +1508,34 @@ func getNodeExternalIP(node *api.Node) string {
 	return "<none>"
 }
 
+// findNodeRole returns the role of a given node, or "" if none found.
+// The role is determined by looking in order for:
+// * a kubernetes.io/role label
+// * a kubeadm.alpha.kubernetes.io/role label
+// * a taint with Key 'dedicated'
+// If no role is found, ("") is returned
+func findNodeRole(node *api.Node) string {
+	if role := node.Labels["kubernetes.io/role"]; role != "" {
+		return role
+	}
+	if role := node.Labels["kubeadm.alpha.kubernetes.io/role"]; role != "" {
+		return role
+	}
+
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		glog.Warningf("error parsing node taints for node %q: %v", node.Name, err)
```

Comment:
This is going to muddy output on the CLI, particularly since the command will still exit 0.  You could change the column content if you wanted.